src/hyp3_sdk/hyp3.py

```diff
@@ -40,17 +41,29 @@ def __init__(
                 Both username and password must be provided if either is provided.
             password: Password for authenticating to `urs.earthdata.nasa.gov`.
                 Both username and password must be provided if either is provided.
-            prompt: Prompt for username and/or password interactively when they
-                are not provided as keyword parameters
+            token: Earth Data Login Bearer Token for authenticating to `urs.earthdata.nasa.gov`
+            prompt: Either 'password' or 'token' to prompt for EDL user name and password or EDL bearer token, respectively.
         """
         self.url = api_url
 
-        if username is None and prompt:
-            username = input('NASA Earthdata Login username: ')
-        if password is None and prompt:
-            password = getpass('NASA Earthdata Login password: ')
+        if prompt and prompt != 'password' and prompt != 'token':
+            warnings.warn(
+                'Passing `prompt=True` is deprecated. Please use either `prompt="password"` or `prompt="token"`',
+                UserWarning,
+            )
+            prompt = 'password'
+
+        if prompt == 'password':
+            if username is None:
+                username = input('NASA Earthdata Login username: ')
+            if password is None:
+                password = getpass('NASA Earthdata Login password: ')
+
+        if prompt == 'token':
+            if token is None:
+                token = getpass('NASA Earthdata Login bearer token: ')
```

Comment:
Maybe something like this, to enforce that only the three expected values are used?
```suggestion
        if prompt is True:
            warnings.warn(
                'Passing `prompt=True` is deprecated. Please use either `prompt="password"` or `prompt="token"`',
                UserWarning,
            )
            prompt = 'password'

        elif prompt == 'password':
            if username is None:
                username = input('NASA Earthdata Login username: ')
            if password is None:
                password = getpass('NASA Earthdata Login password: ')

        elif prompt == 'token':
            if token is None:
                token = getpass('NASA Earthdata Login bearer token: ')
        
        else:
            raise ValueError(f'Unexpected value {prompt} for `prompt`')
```

Reply:
Oh wait, that doesn't quite work, disregard.